api/src/main/java/org/apache/iceberg/actions/RewriteDeletes.java

```diff
@@ -0,0 +1,55 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.iceberg.actions;
+
+import java.util.Set;
+import org.apache.iceberg.DeleteFile;
+
+public interface RewriteDeletes extends SnapshotUpdate<RewriteDeletes, RewriteDeletes.Result> {
+
+  /**
+   * rewrite the equality deletes.
+   *
+   * @return this for method chaining
+   */
+  RewriteDeletes rewriteEqDeletes();
```

Comment:
Curious on the reason for having `rewriteEqDeletes` and `rewritePosDeletes` here, looks like in `RewriteDataFiles` we define methods based on the exact strategy to use, and for rewriting delete files there could be strategies like eqDelete->posDelete, combine posDelete into one, and even take both delete and data files and return data files (I think the last one should be a separate interface). Here it seems like the user can decide whether to rewrite certain type of delete files, but I would think strategy might be the thing that the user want to control more? Sorry I've been away from the community conversations for a while so might lack a lot of context if this was previously discussed.

Reply:
`rewriteEqDeletes` and `rewritePosDeletes` is a top-level choice for users to choose which kind of delete he or she wants to rewrite.  Here's an example: `actionsProvider().rewriteDeletes(table).rewriteEqDeletes().execute();`, users could clearly know what kind of deletes are handled. But you remind me that, we might also don't need these two as well, use one `strategy` API to select its implementation is enough.

The strategy options are open and we could pass options from `options` interface from the specified strategy implementation.